packages/publish-views/lib/loadViews.js

```diff
@@ -38,13 +41,27 @@ export default async function loadViewsToPublish() {
   }))
 }
 
-async function loadViewMeta(view, client) {
+async function loadViewMeta(publishedView, metaObject, client) {
   const shape = await viewShape()
 
   const subjectVariable = 'view'
-  const query = shapeTo.construct(shape, { focusNode: view, subjectVariable })
+  const query = shapeTo.construct(shape, { focusNode: metaObject, subjectVariable })
+  const metaStream = await query.execute(client.query)
+
+  return metaStream.pipe(through2.obj(viewIdTransform(metaObject, publishedView)))
+}
 
-  return query.execute(client.query)
+function viewIdTransform(from, to) {
+  const rename = term => (term.equals(from) ? to : from)
+
+  return function renameToPublishedView(quad, cb, next) {
+    const subject = rename(quad.subject)
+    const predicate = rename(quad.predicate)
+    const object = rename(quad.object)
+
+    this.push($rdf.quad(subject, predicate, object))
+    next()
+  }
```

Comment:
Because we no longer assume the same identify of the metadata object and the published view, the constructed data needs to be transformed